### ISSUE TITLE
feature request: async function as nuxt.config

### ISSUE THREAD

**Issue by TasukuUno** [nuxt/framework#1611]
### Environment

- Operating System: `Darwin`
- Node Version:     `v16.12.0`
- Nuxt Version:     `3.0.0-27258467.4e424d0`
- Package Manager:  `npm@8.1.0`
- Bundler:          `Vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Describe the bug

I would like to define nuxt configs by executing an async function as is possible in Nuxt v2..!!
It is useful to fetch and set the configs asynchronously.

```ts
import { defineNuxtConfig } from "nuxt3";

export default async () => {
  const publicRuntimeConfig = await fetchConfigs();

  return defineNuxtConfig({
    publicRuntimeConfig
  });
};

async function fetchConfigs() {
  await new Promise((resolve) => setTimeout(resolve, 1000));

  return {
    MESSAGE: "HELLO WORLD"
  };
}
```

Is it acceptable for you?

### Reproduction

https://codesandbox.io/s/nuxt3-config-function-sqdep?file=/nuxt.config.ts

### Additional context

I think these are related codes.

### v2

https://github.com/nuxt/nuxt.js/blob/db11f7bf937cdbb96ff0067bdee1aa153978e4e0/packages/config/src/load.js#L61-L71

### v3

https://github.com/nuxt/framework/blob/d6ad4f3b16d124f01b21bb535a8e224020526cfb/packages/kit/src/config/load.ts#L34-L39

**Comment by danielroe** [comment]
Would you be able to use an async buildModule that updates/changes the configuration to achieve the same result?

**Comment by TasukuUno** [comment]
@danielroe Hi, thanks for your reply!!

> use an async buildModule that updates/changes the configuration

I had never tried that approach, so I wrote the following example.
https://codesandbox.io/s/nuxt3-define-runtime-configs-in-modules-wv434?file=/nuxt.config.ts

Interestingly, I faced that I can't extend publicRuntimeConfig in buildModules....!!
Is it because `initNitro()` is already done? Or are there any hook I can use?

https://github.com/nuxt/framework/blob/d6ad4f3b16d124f01b21bb535a8e224020526cfb/packages/nuxt3/src/core/nuxt.ts#L39-L53
